pkg/alertmanager/amcfg.go

```diff
@@ -0,0 +1,509 @@
+	SMTPAuthSecret   string          `yaml:"smtp_auth_secret,omitempty" json:"smtp_auth_secret,omitempty"`
+	SMTPAuthIdentity string          `yaml:"smtp_auth_identity,omitempty" json:"smtp_auth_identity,omitempty"`
+	SMTPRequireTLS   bool            `yaml:"smtp_require_tls,omitempty" json:"smtp_require_tls,omitempty"`
+	SlackAPIURL      *config.URL     `yaml:"slack_api_url,omitempty" json:"slack_api_url,omitempty"`
+	PagerdutyURL     *config.URL     `yaml:"pagerduty_url,omitempty" json:"pagerduty_url,omitempty"`
+	HipchatAPIURL    *config.URL     `yaml:"hipchat_api_url,omitempty" json:"hipchat_api_url,omitempty"`
+	HipchatAuthToken string          `yaml:"hipchat_auth_token,omitempty" json:"hipchat_auth_token,omitempty"`
+	OpsGenieAPIURL   *config.URL     `yaml:"opsgenie_api_url,omitempty" json:"opsgenie_api_url,omitempty"`
+	OpsGenieAPIKey   string          `yaml:"opsgenie_api_key,omitempty" json:"opsgenie_api_key,omitempty"`
+	WeChatAPIURL     *config.URL     `yaml:"wechat_api_url,omitempty" json:"wechat_api_url,omitempty"`
+	WeChatAPISecret  string          `yaml:"wechat_api_secret,omitempty" json:"wechat_api_secret,omitempty"`
+	WeChatAPICorpID  string          `yaml:"wechat_api_corp_id,omitempty" json:"wechat_api_corp_id,omitempty"`
+	VictorOpsAPIURL  *config.URL     `yaml:"victorops_api_url,omitempty" json:"victorops_api_url,omitempty"`
+	VictorOpsAPIKey  string          `yaml:"victorops_api_key,omitempty" json:"victorops_api_key,omitempty"`
+}
+
+type route struct {
+	Receiver       string            `yaml:"receiver,omitempty" json:"receiver,omitempty"`
+	GroupByStr     []string          `yaml:"group_by,omitempty" json:"group_by,omitempty"`
+	Match          map[string]string `yaml:"match,omitempty" json:"match,omitempty"`
+	MatchRE        map[string]string `yaml:"match_re,omitempty" json:"match_re,omitempty"`
+	Continue       bool              `yaml:"continue,omitempty" json:"continue,omitempty"`
+	Routes         []*route          `yaml:"routes,omitempty" json:"routes,omitempty"`
+	GroupWait      string            `yaml:"group_wait,omitempty" json:"group_wait,omitempty"`
+	GroupInterval  string            `yaml:"group_interval,omitempty" json:"group_interval,omitempty"`
+	RepeatInterval string            `yaml:"repeat_interval,omitempty" json:"repeat_interval,omitempty"`
+}
+
+type inhibitRule struct {
+	TargetMatch   map[string]string `yaml:"target_match,omitempty" json:"target_match,omitempty"`
+	TargetMatchRE map[string]string `yaml:"target_match_re,omitempty" json:"target_match_re,omitempty"`
+	SourceMatch   map[string]string `yaml:"source_match,omitempty" json:"source_match,omitempty"`
+	SourceMatchRE map[string]string `yaml:"source_match_re,omitempty" json:"source_match_re,omitempty"`
+	Equal         []string          `yaml:"equal,omitempty" json:"equal,omitempty"`
+}
+
+type receiver struct {
+	Name string `yaml:"name" json:"name"`
+	// EmailConfigs     []*EmailConfig     `yaml:"email_configs,omitempty" json:"email_configs,omitempty"`
```

Comment:
Lets leave a TODO here why they are commented out.

Reply:
I think the created issues are better than a TODO, so I'll just remove these commented out lines here instead.